src/rules/memberAccessRule.ts

```diff
@@ -15,127 +15,123 @@
         rationale: "Explicit visibility declarations can make code more readable and accessible for those new to TS.",
         optionsDescription: Lint.Utils.dedent`
-            Two arguments may be optionally provided:
+            These arguments may be optionally provided:
 
-            * \`"check-accessor"\` enforces explicit visibility on get/set accessors (can only be public)
-            * \`"check-constructor"\`  enforces explicit visibility on constructors (can only be public)`,
+            * \`"no-public"\` forbids public accessibility to be specified, because this is the default.
+            * \`"check-accessor"\` enforces explicit visibility on get/set accessors
+            * \`"check-constructor"\`  enforces explicit visibility on constructors`,
         options: {
             type: "array",
             items: {
                 type: "string",
-                enum: ["check-accessor", "check-constructor"],
+                enum: [OPTION_NO_PUBLIC, OPTION_CHECK_ACCESSOR, OPTION_CHECK_CONSTRUCTOR],
             },
             minLength: 0,
-            maxLength: 2,
+            maxLength: 3,
         },
-        optionExamples: ["true", '[true, "check-accessor"]'],
+        optionExamples: ["true", `[true, "${OPTION_NO_PUBLIC}"]`, `[true, "${OPTION_CHECK_ACCESSOR}"]`],
         type: "typescript",
         typescriptOnly: true,
     };
     /* tslint:enable:object-literal-sort-keys */
 
-    public static FAILURE_STRING_FACTORY = (memberType: string, memberName: string | undefined, publicOnly: boolean) => {
+    public static FAILURE_STRING_NO_PUBLIC = "'public' is implicit.";
+
+    public static FAILURE_STRING_FACTORY(memberType: string, memberName: string | undefined): string {
         memberName = memberName === undefined ? "" : ` '${memberName}'`;
-        if (publicOnly) {
-            return `The ${memberType}${memberName} must be marked as 'public'`;
-        }
         return `The ${memberType}${memberName} must be marked either 'private', 'public', or 'protected'`;
     }
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithWalker(new MemberAccessWalker(sourceFile, this.getOptions()));
-    }
-}
-
-export class MemberAccessWalker extends Lint.RuleWalker {
-    public visitConstructorDeclaration(node: ts.ConstructorDeclaration) {
-        if (this.hasOption("check-constructor")) {
-            // constructor is only allowed to have public or nothing, but the compiler will catch this
-            this.validateVisibilityModifiers(node);
+        const options = this.getOptions().ruleArguments;
+        const noPublic = options.indexOf(OPTION_NO_PUBLIC) !== -1;
+        let checkAccessor = options.indexOf(OPTION_CHECK_ACCESSOR) !== -1;
+        let checkConstructor = options.indexOf(OPTION_CHECK_CONSTRUCTOR) !== -1;
+        if (noPublic) {
+            if (checkAccessor || checkConstructor) {
```

Comment:
This checks that when `no-public` is set, `check-accessor` and `check-constructor` aren't set, because someone who wants to ban the `public` keyword presumably wants to ban it everywhere, so we always check everything, so other options aren't necessary.